bundles/org.openhab.binding.teslapowerwall/src/main/java/org/openhab/binding/teslapowerwall/internal/TeslaPowerwallWebTargets.java

```diff
@@ -0,0 +1,246 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.teslapowerwall.internal;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Properties;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.teslapowerwall.internal.api.BatterySOE;
+import org.openhab.binding.teslapowerwall.internal.api.GridStatus;
+import org.openhab.binding.teslapowerwall.internal.api.MeterAggregates;
+import org.openhab.binding.teslapowerwall.internal.api.Operations;
+import org.openhab.binding.teslapowerwall.internal.api.SystemStatus;
+import org.openhab.core.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * Handles performing the actual HTTP requests for communicating with Tesla Powerwall units.
+ *
+ * @author Paul Smedley - Initial Contribution
+ *
+ */
+@NonNullByDefault
+public class TeslaPowerwallWebTargets {
+    private static final int TIMEOUT_MS = 30000;
+
+    private String getBatterySOEUri;
+    private String getGridStatusUri;
+    private String getSystemStatusUri;
+    private String getMeterAggregatesUri;
+    private String getTokenUri;
+    private String getOperationUri;
+    private String setOperationUri;
+    private final Logger logger = LoggerFactory.getLogger(TeslaPowerwallWebTargets.class);
+
+    public TeslaPowerwallWebTargets(String ipAddress) {
+        String baseUri = "https://" + ipAddress + "/";
+        getBatterySOEUri = baseUri + "api/system_status/soe";
+        getGridStatusUri = baseUri + "api/system_status/grid_status";
+        getSystemStatusUri = baseUri + "api/system_status";
+        getMeterAggregatesUri = baseUri + "api/meters/aggregates";
+        getTokenUri = baseUri + "api/login/Basic";
+        getOperationUri = baseUri + "api/operation";
+        setOperationUri = baseUri + "api/operation";
+    }
+
+    public BatterySOE getBatterySOE(String token) throws TeslaPowerwallCommunicationException {
+        String response;
+        Properties headers = new Properties();
+        headers.setProperty("Content-Type", token);
+
+        synchronized (this) {
+            try {
+                response = HttpUtil.executeUrl("GET", getBatterySOEUri, headers, null, null, TIMEOUT_MS);
+            } catch (IOException ex) {
+                logger.debug("{}", ex.getLocalizedMessage(), ex);
+                // Response will also be set to null if parsing in executeUrl fails so we use null here to make the
+                // error check below consistent.
+                response = null;
+            }
+        }
+
+        if (response == null) {
+            throw new TeslaPowerwallCommunicationException(
+                    String.format("Tesla Powerwall returned error while invoking %s", getBatterySOEUri));
+        }
+        logger.debug("getBatterySOE response = {}", response);
```

Comment:
```suggestion
        logger.trace("getBatterySOE response = {}", response);
```
request metadata can be logged to debug, content should be trace.
For reference: https://www.openhab.org/docs/developer/guidelines.html#f-logging

Also applies to other http calls in this binding

Reply:
Fixed